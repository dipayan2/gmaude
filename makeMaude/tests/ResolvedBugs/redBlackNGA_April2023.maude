set show timing off .
***
***	Test that red-black ACU stripper-collector full matcher isn't run if there is an non-ground alien.
***

fmod FOO is
  pr NAT .
  sort  Set .
  subsort Nat < Set .
var N : Nat .
  op f : Set Set -> Set [assoc comm] .
  op g : Nat -> Set .
  eq g(0) = 0 .
  eq g(s N) = f(g(N), s N) .

var S : Set .
var M : Nat .
  op h : Set -> Set .
  op i : Nat -> Nat .
  ceq h(f(N, S, i(M))) = 0 if N > 0 .
endfm

red h(g(7)) .
red h(g(8)) .

fmod FOO is
  pr NAT .
  sort  Set .
  subsort Nat < Set .
var N : Nat .
  op f : Set Set -> Set [assoc comm] .
  op g : Nat -> Set .
  eq g(0) = 0 .
  eq g(s N) = f(g(N), s N) .

var S : Set .
var M : Nat .
  op h : Set -> Set .
  op i : Nat -> Nat .
  ceq h(f(N, S, i(M))) = M if N > 0 .
endfm

red h(g(8)) .
