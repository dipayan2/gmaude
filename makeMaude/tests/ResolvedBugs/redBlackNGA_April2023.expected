==========================================
reduce in FOO : h(g(7)) .
rewrites: 8
result Set: h(f(0, 1, 2, 3, 4, 5, 6, 7))
==========================================
reduce in FOO : h(g(8)) .
rewrites: 9
result Set: h(f(0, 1, 2, 3, 4, 5, 6, 7, 8))
==========================================
reduce in FOO : h(g(8)) .
rewrites: 9
result Set: h(f(0, 1, 2, 3, 4, 5, 6, 7, 8))
Bye.
